pkg/internal/infrastructure/terraform.go

```diff
@@ -87,6 +92,20 @@ func ComputeTerraformerChartValues(
 		if config.Networks.CloudNAT.MinPortsPerVM != nil {
 			minPortsPerVM = *config.Networks.CloudNAT.MinPortsPerVM
 		}
+		if config.Networks.CloudNAT.NatIPNames != nil {
+			natIPNames := []string{}
+			nIPs := []string{}
+			natIPOutputList := []string{}
+			for _, v := range config.Networks.CloudNAT.NatIPNames {
+				natIPNames = append(natIPNames, v.Name)
+				nIPs = append(nIPs, fmt.Sprintf("\"${data.google_compute_address.%s.self_link}\"", v.Name))
+				natIPOutputList = append(natIPOutputList, fmt.Sprintf("${data.google_compute_address.%s.address}", v.Name))
+			}
+			cN["natIPs"] = fmt.Sprintf("[%s]", strings.Join(nIPs[:], ","))
+			cN["natIPNames"] = natIPNames
+			cN["natIPOutput"] = strings.Join(natIPOutputList[:], ",")
+			cN["minPortsPerVM"] = minPortsPerVM
```

Comment:
Why is this inside the `if config.Networks.CloudNAT.NatIPNames != nil {` condition? It is completely independent from the nat IPs, or? Move it out?

Reply:
Why is this independent? If nat ip names are specified those values have to be calculated.

Reply:
I don't see any specific calculation for the min ports per VM in case the NAT IP names are set. Also, the min ports are already assigned here https://github.com/gardener/gardener-extension-provider-gcp/pull/39/files#diff-4787a75463b112a387ec6927df20e486R144